src/main/java/com/greenfox/dramacsoport/petclinicbackend/services/petHandling/PetServiceImpl.java

```diff
@@ -30,4 +30,12 @@ public PetListResponse getUserPets(String email) {
 
         return new PetListResponse(petDTOList);
     }
+
+    @Override
+    public Pet addPet(String email, PetDTO petDTO) {
+        Pet pet = modelMapper.map(petDTO, Pet.class);
+        pet.setOwner(appUserAuthService.loadUserByUsername(email));
+        petRepository.save(pet);
+        return pet;
```

Comment:
Oh well, I did not see last time this. Sorry for noting in another round: you probably should not return (in any case) with an entity. You are risking exposing fields, which are not used in the clients and you are taking higher risk of circular json references https://medium.com/@raksmeykoung_19675/avoiding-circular-references-or-cycles-spring-data-jpa-35ca137ad3cb
. Please use DTOs, when reading request-bodies and returning response-bodies too.

Reply:
oh indeed.. thanks, I corrected it.